thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java

```diff
@@ -204,6 +218,36 @@ public Long getMetricMaxDataTime(@PathParam("metricId") Long metricId) {
   }
 
   @GET
+  @Path("autocomplete/function")
+  public List<AnomalyFunctionDTO> getFunctionsWhereNameLike(@QueryParam("name") String name) {
+    List<AnomalyFunctionDTO> functions = Collections.emptyList();
+    if (StringUtils.isNotBlank(name)) {
+      functions = anomalyFunctionDAO.findWhereNameLike("%" + name + "%");
+    }
+    return functions;
+  }
+
+  @GET
```

Comment:
Can you add some comments here about the usage and parameters. 
It is little bit confusing here -- you have both name and appName as query parameters.